src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1120,6 +1135,15 @@ public static void mutateAtomically(Collection<Mutation> mutations,
                                         ClientState clientState)
     throws UnavailableException, OverloadedException, WriteTimeoutException
     {
+        atomicMutator.compareAndSet(null, DEFAULT_ATOMIC_MUTATOR);
```

Comment:
I think we should avoid this in the hot path..

Reply:
why?
I mean - CAS is not free, but `LOCK XCHG` doesn't cost that much.
We already have locks e.g. on writing to memtable (per single partition update). 
Here we've got a single CAS for multiple mutations.
Not to mention that write is synchronous so it easily dominates any CAS.